repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/AuthPolicyPreProcessor.java

```diff
@@ -165,6 +165,36 @@ private void processCreatePolicy(AtlasStruct entity) throws AtlasBaseException {
         RequestContext.get().endMetricRecord(metricRecorder);
     }
 
+    private void validateAndReduce(AtlasEntity policy) {
+        ArrayList<String> resources = (ArrayList<String>) policy.getAttribute(ATTR_POLICY_RESOURCES);
+        boolean hasAllDomainPattern = false;
+
+        for (int i = 0; i < resources.size(); i++) {
+            String[] resourceParts = resources.get(i).split(":");
+            String resource = resourceParts[resourceParts.length - 1];
+            if (resource.equals("*") || resource.equals("*/super") || resource.equals("default/domain/*/super")) {
+                resources.remove(i);
+                hasAllDomainPattern = true;
+                i--;
+            }
+        }
+
+        if (hasAllDomainPattern) {
+            for (int i = 0; i < resources.size(); i++) {
+                String[] resourceParts = resources.get(i).split(":");
+                String resource = resourceParts[resourceParts.length - 1];
+                if (resource.matches("default/domain/.+/super(/.+)?")) {
+                    resources.remove(i);
+                    i--;
+                }
+            }
+            resources.add(ENTITY_DEFAULT_DOMAIN_SUPER);
```

Comment:
this is not needed
if (hasAllDomainPattern) { 
    policy.setAttribute(ATTR_POLICY_RESOURCES, Arrays.asList("entity:default/domain/*/super"));
}

Reply:
Ok , but my concern is what if we send other entities as well in policy, not related to datadomain, they will be also be replaced.
Adding the requested change regardless for now.

Reply:
Given that we do this only when policySubCategory is `domain`, we are 100% sure that there must not be any other entities in the policy.
With this, when users select All Domains, it will not create any issue as policy will be applicable to all domains & assets inside all domain based on selected permissions